cookbook/form/data_transformers.rst

```diff
@@ -434,23 +397,69 @@ Now, whenever you need to use your special ``issue_selector`` field type,
 it's quite easy::
 
     // src/AppBundle/Form/TaskType.php
-    namespace AppBundle\Form;
-
-    use Symfony\Component\Form\AbstractType;
-    use Symfony\Component\Form\FormBuilderInterface;
+    use AppBundle\Form\DataTransformer\IssueToNumberTransformer;
 
     class TaskType extends AbstractType
     {
         public function buildForm(FormBuilderInterface $builder, array $options)
         {
             $builder
-                ->add('task')
-                ->add('dueDate', null, array('widget' => 'single_text'))
+                ->add('description', 'textarea')
                 ->add('issue', 'issue_selector');
         }
 
-        public function getName()
-        {
-            return 'task';
-        }
+        // ...
     }
+
+About Model and View Transformers
+---------------------------------
+
+In the above example, the transformer was used as a "model" transformer.
+In fact, there are two different types of transformers and three different
+types of underlying data.
+
+.. image:: /images/cookbook/form/DataTransformersTypes.png
+   :align: center
+
+In any form, the three different types of data are:
+
+1) **Model data** - This is the data in the format used in your application
+   (e.g. an ``Issue`` object). If you call ``Form::getData`` or ``Form::setData``,
+   you're dealing with the "model" data.
+
+2) **Norm Data** - This is a normalized version of your data, and is commonly
+   the same as your "model" data (though not in our example). It's not commonly
+   used directly.
+
+3) **View Data** - This is the format that's used to fill in the form fields
+   themselves. It's also the format in which the user will submit the data. When
+   you call ``Form::submit($data)``, the ``$data`` is in the "view" data format.
+
+The two different types of transformers help convert to and from each of these
+types of data:
+
+**Model transformers**:
+    - ``transform``: "model data" => "norm data"
+    - ``reverseTransform``: "norm data" => "model data"
+
+**View transformers**:
+    - ``transform``: "norm data" => "view data"
+    - ``reverseTransform``: "view data" => "norm data"
+
+Which transformer you need depends on your situation.
+
+To use the view transformer, call ``addViewTransformer``.
+
+So why Use the Model Transformer?
+---------------------------------
+
+In this example, the field is a ``text`` field, and a text field is always
+expected to be a simple, scalar format in the "norm" and "view" formats. For
+this reason, the most appropriate transformer was the "model" transformer
+(which converts to/from the *norm* format - string issue number - to the *model*
+format - Issue object).
```

Comment:
in this case, a model transformer should indeed be used because of this other rule:

when extending another form type, the parent decides what the norm data is. You can only make it more specific (the ``form`` type does not care about the norm data and accepts anything, so you don't really need to consider this one)